tasks/src/main/scala/sbt/TaskId.scala

```diff
@@ -0,0 +1,6 @@
+package sbt
+
+import sbt.internal.util.AttributeMap
+
+trait TaskId[A]:
+  def tags: ConcurrentRestrictions.TagMap
```

Comment:
Interesting. So at the end of the day the information it wants besides the identity is tags?

Reply:
Yes, the tags are used by the `ConcurrentRestrictions` implementations. It was a `F[Any] => TagMap` parameter and now it is a field in `TaskId[A]`.